documentation/asciidoc/computers/linux_kernel/building.adoc

```diff
@@ -1,342 +1,462 @@
-make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- bcmrpi_defconfig
+$ cd linux
+$ KERNEL=kernel
+$ make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- bcmrpi_defconfig
 ----
+| Raspberry Pi Zero
+| Raspberry Pi Zero W
+| Raspberry Pi Compute Module 1
 
-For Raspberry Pi 2, 3, 3+ and Zero 2 W, and Raspberry Pi Compute Modules 3 and 3+:
-
-[,bash]
+| Raspberry Pi 2
+.6+.^|
+[source,console]
 ----
-cd linux
-KERNEL=kernel7
-make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- bcm2709_defconfig
+$ cd linux
+$ KERNEL=kernel7
+$ make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- bcm2709_defconfig
 ----
+| Raspberry Pi 3
+| Raspberry Pi 3+
+| Raspberry Pi Zero 2 W
+| Raspberry Pi Compute Module 3
+| Raspberry Pi 3+:
 
-For Raspberry Pi 4 and 400, and Raspberry Pi Compute Module 4:
-
-[,bash]
+| Raspberry Pi 4
+.4+.^|
+[source,console]
 ----
-cd linux
-KERNEL=kernel7l
-make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- bcm2711_defconfig
+$ cd linux
+$ KERNEL=kernel7l
+$ make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- bcm2711_defconfig
 ----
+| Raspberry Pi 400
+| Raspberry Pi Compute Module 4
+| Raspberry Pi Compute Module 4S
 
-===== 64-bit configs
-
-For Raspberry Pi 3, 3+, 4, 400 and Zero 2 W, and Raspberry Pi Compute Modules 3, 3+ and 4:
 
-[,bash]
+.8+^.^| 64-bit
+| Raspberry Pi 3
+.8+.^| [source,console]
 ----
-cd linux
-KERNEL=kernel8
-make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- bcm2711_defconfig
+$ cd linux
+$ KERNEL=kernel8
+$ make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- bcm2711_defconfig
 ----
+| Raspberry Pi 3+
+| Raspberry Pi 4
+| Raspberry Pi 400
+| Raspberry Pi Zero 2 W
+| Raspberry Pi Compute Modules 3
+| Raspberry Pi Compute Module 4
+| Raspberry Pi Compute Module 4S
 
-For Raspberry Pi 5:
-
-[,bash]
+| Raspberry Pi 5
+.1+.^|
+[source,console]
 ----
-cd linux
-KERNEL=kernel_2712
-make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- bcm2712_defconfig
+$ cd linux
+$ KERNEL=kernel_2712
+$ make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- bcm2712_defconfig
 ----
+|===
 
-NOTE: The standard, `bcm2711_defconfig`-based kernel (`kernel8.img`) also runs on Raspberry Pi 5. For best performance you should use `kernel_2712.img`, but for situations where a 4KB page size is required then `kernel8.img` (`kernel=kernel8.img`) should be used.
-
-===== Build with configs
-
-NOTE: To speed up compilation on multiprocessor systems, and get some improvement on single-processor devices, use `-j n`, where n is the number of processors × 1.5. You can use the `nproc` command to see how many processors you have. 
+==== Build with modules and Device Tree
 
-====== For all 32-bit builds
+For 64-bit kernels, pass the image modules and DTB files like so:
 
-[,bash]
+[source,console]
 ----
-make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- zImage modules dtbs
+$ make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- Image modules dtbs
 ----
 
-====== For all 64-bit builds
+For 32-bit kernels, pass the image modules and DTB files like so, prefixing each image module with `z`:
```

Comment:
Should this comment be added to the native-build section too? OR perhaps this comment should be removed from here?
(Either way, "each image module" isn't the right terminology to use - there's only one module so "each" isn't necessary/appropriate, and `zImage` is the Linux kernel, not a "module" :wink: )

Reply:
Trimmed a bunch of prose that wasn't that accurate, thanks!